### ISSUE TITLE
Use sidecar templates for icon components

### ISSUE THREAD

**Issue by barmintor** [projectblacklight/blacklight#2723]
On the committers' call @hackartisan asked whether we could leverage the limited override of sidecar templates from `Blacklight::Component` to provide a way to get inline SVG from the icon components without having the svg source in Ruby.

One way to do this would be to have a raw template.

See also conversation on #2670

**Comment by jcoyne** [comment]
@barmintor yeah, we could do that. I only made these class variable configuration as that seemed more "APIish".   I'm not convinced that "just override a template" is a very good pattern in that it doesn't allow blacklight to evolve without breaking things in peoples apps.

**Comment by barmintor** [comment]
Also good points! But I want to capture the ideas that came up in the call.

**Comment by barmintor** [comment]
This _works_, but I'm not sure it's better. Something to think about.

**Comment by barmintor** [comment]
@jcoyne I also think, to your point, that we need to document that the component templates are private API. I think the compiler override is a necessary bridge, but it shouldn't be recommended practice.
